app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,36 @@
+class CommentsController < ApplicationController
+  before_action :ensure_logged_in, only:[:create]
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    if (params.has_key?(:answer_id))
```

Comment:
If your routes are nested a trick like the following can be helpful to avoid a bunch of if / elseif stuff.

```
private
  def load_voteable
    resource, id = request.path.split('/')[1, 2]
    @voteable = resource.singularize.classify.constantize.find(id)
  end
```